### PR TITLE
ISPN-11507 Commons Split (and fibers!)

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-10507
ISPN-11507 Split commons into multiple modules

https://issues.redhat.com/browse/ISPN-10508
ISPN-11508 JDK-specific utility class to retrieve process information

https://issues.redhat.com/browse/ISPN-10509
ISPN-11509 Project Loom fibers

### CODE REVIEW

File: commons/loom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      <groupId>org.infinispan</groupId>
+      <artifactId>infinispan-commons-parent</artifactId>
+      <version>11.0.0-SNAPSHOT</version>
+      <relativePath>../pom.xml</relativePath>
+   </parent>
+
+   <artifactId>infinispan-commons-loom</artifactId>
+   <packaging>jar</packaging>
+   <name>Infinispan Commons JDK Loom</name>
+   <description>Infinispan Commons JDK 11</description>
```
Comment (pruivo) [inline commons/loom/pom.xml]: typo
```suggestion
   <description>Infinispan Commons JDK Loom</description>
```

File: pom.xml [hunk 2]
```diff
@@ -2374,6 +2374,16 @@
             <scope>test</scope>
             <version>${version.infinispan}</version>
          </dependency>
+         <dependency>
+            <groupId>org.infinispan</groupId>
+            <artifactId>infinispan-commons-base</artifactId>
```
Comment (pruivo) [inline pom.xml]: where is commons-base defined?

File: commons/loom/pom.xml (hunk 1 above)
Comment (tristantarrant) [reply commons/loom/pom.xml]: Fixed

File: pom.xml (hunk 2 above)
Comment (tristantarrant) [reply pom.xml]: That was a leftover from a previous iteration. Removed

File: commons/all/src/main/assemblies/mr.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+   <id>mr</id>
+   <formats>
+      <format>jar</format>
+   </formats>
+   <includeBaseDirectory>false</includeBaseDirectory>
+   <moduleSets>
+      <moduleSet>
+         <useAllReactorProjects>true</useAllReactorProjects>
+         <includes>
+            <include>org.infinispan:infinispan-commons-base</include>
```
Comment (pruivo) [inline commons/all/src/main/assemblies/mr.xml]: so... this file can be removed as well? didn't find any reference to it.

**Comment by pruivo** [comment]
are you able to run the core testsuite with "JDK Loom"? 
I'm getting this error:
```
[Byte Buddy] ERROR java.util.stream.MatchOps [null, module java.base, loaded=false]
java.lang.IllegalArgumentException: Unsupported class file major version 59
	at reactor.blockhound.shaded.net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:195)
	at reactor.blockhound.shaded.net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:176)
	at reactor.blockhound.shaded.net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:162)
```

Version
```
$ mvn -version
Apache Maven 3.6.2 (40f52333136460af0dc0d7232c0dc0bcf0d9e117; 2019-08-27T16:06:16+01:00)
Maven home: /mnt/data/pedro/opt/maven
Java version: 15-loom, vendor: Oracle Corporation, runtime: /mnt/data/pedro/opt/jdk-15-loom
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.5.8-1-manjaro", arch: "amd64", family: "unix"
```

**Comment by tristantarrant** [comment]
We need a new release of BlockHound with bytebuddy 1.10.8. @bsideup @wburns

File: commons/all/src/main/assemblies/mr.xml (hunk 3 above)
Comment (tristantarrant) [reply commons/all/src/main/assemblies/mr.xml]: removed it.

**Comment by bsideup** [comment]
@tristantarrant thanks for pinging me! BlockHound 1.0.3 will be released very soon, probably today. Meanwhile, you can try the latest SNAPSHOT.

**Comment by pruivo** [comment]
It works with BlockHound `1.0.3.BUILD-SNAPSHOT` 
I have 12 tests failing with `java.lang.NoClassDefFoundError: java/security/acl/Group`. We can fix them later since JDK15 isn't released yet.

**Comment by tristantarrant** [comment]
I need to kill that use of java.security.acl.Group asap

**Comment by tristantarrant** [comment]
https://github.com/infinispan/infinispan/pull/8083

**Comment by pruivo** [comment]
integrated! thanks @tristantarrant !

**Pull request merged**